### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by fpob**
* fix API spec and enable strict validation and response validation
* fix some warnings from SQLAlchemy
* add missing authorization in tower and auth modules
* ... see commits

### CODE REVIEW

File: src/rhub/api/__init__.py [hunk 1]
```diff
@@ -62,11 +56,8 @@ def get_vault() -> Vault:
 
 def init_app():
     logger.info('Starting inititialization...')
```
Comment (sai-c) [inline src/rhub/api/__init__.py:58]: Not a big deal but "inititialization" -> "initialization"

File: src/rhub/api/_setup.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        keycloak.user_create({
+            'username': ADMIN_USER,
+            'email': 'nobody@redhat.com',
+            'firstName': 'Admin',
+            'lastName': 'RHub',
+        })
+
+    admin_user = keycloak.user_list({'username': ADMIN_USER})[0]
+    if not any(group['name'] == ADMIN_GROUP
+               for group in keycloak.user_group_list(admin_user['id'])):
```
Comment (sai-c) [inline src/rhub/api/_setup.py:50]: Spacing seems off (or is that just because of GitHub)?

File: src/rhub/api/__init__.py (hunk 1 above)
Comment (sai-c) [inline src/rhub/api/__init__.py:58]: Not a big deal but "inititialization" -> "initialization"

File: src/rhub/api/tower.py [hunk 3]
```diff
@@ -113,10 +118,11 @@ def get_template(template_id):
 
     except Exception as e:
         logger.exception(e)
-        return problem(500, 'Server Error', 'Uknown server error, {e}')
+        return problem(500, 'Server Error', f'Uknown server error, {e}')
```
Comment (sai-c) [inline src/rhub/api/tower.py:121]: Again, a super small thing which probably doesn't matter, but "Uknown" -> "Unknown"

File: src/rhub/api/__init__.py [hunk 4]
```diff
@@ -62,11 +56,8 @@ def get_vault() -> Vault:
...
-    for name in MODULES:
-        module = importlib.import_module(f'rhub.{name}')
-        if hasattr(module, 'init'):
-            logger.info(f'Initializing "{name}" module...')
-            module.init()
+    from ._setup import setup
```
Comment (sai-c) [inline src/rhub/api/__init__.py:59]: Is the import okay here?

File: src/rhub/api/_setup.py (hunk 2 above)
Comment (fpob) [reply src/rhub/api/_setup.py:50]: What do you mean? It looks ok on GitHub and flake8 is not complaining.

File: src/rhub/api/__init__.py (hunk 4 above)
Comment (fpob) [reply src/rhub/api/__init__.py:59]: I had some problems with circular imports when it was on top of the file. `_setup` module is used only in init so I guess it is ok in this case.

File: src/rhub/api/tower.py (hunk 3 above)
Comment (fpob) [reply src/rhub/api/tower.py]: It does matter. If you notice a typo, report it. :+1:

File: src/rhub/api/_setup.py (hunk 2 above)
Comment (brhatiga) [reply src/rhub/api/_setup.py:50]: @fpob  this is strange, I just cloned fpob:fixes and it looks wrong there too.  It looks just like what @sai-c posted above.

Comment (fpob) [reply src/rhub/api/_setup.py:50]: I have no idea what you see :smile: 

strange-indent

^ do you see something else?

Comment (sai-c) [reply src/rhub/api/_setup.py:50]: Oh, I actually do see the same thing, but I guess I missed that the for loop was inside the any().

Comment (brhatiga) [reply src/rhub/api/_setup.py:50]: Ah, see it now, the 'for' statement is part of the 'if' statement...missed the colon at the end.

**Review by brhatiga — APPROVED** [review]
lgtm

**Review by sai-c — APPROVED** [review]

**Pull request merged**